tailcfg/tailcfg.go

```diff
@@ -584,6 +591,12 @@ type Hostinfo struct {
 	Cloud           string         `json:",omitempty"`
 	Userspace       opt.Bool       `json:",omitempty"` // if the client is running in userspace (netstack) mode
 	UserspaceRouter opt.Bool       `json:",omitempty"` // if the client's subnet router is running in userspace (netstack) mode
+	Location        *Location      `json:",omitempty"`
```

Comment:
docs

Reply:
Is it necessary to have fields be exclusively one or the other?

In the initial use case, control provides these fields, but there are other fairly obvious use cases where users may want to set them to make use of the same client capabilities.

That is, if the client supports "pick by location, biased by priority", it's reasonable to enable users who manage their own nodes be able to make use of that feature set, in order for them to do so it's reasonable for them to use these fields - unless we have or need a separation based on where data can originate, in which case we need two sets of fields for the data, and to rationalize how they're used.

Reply:
Sure. here's fine if you think we might send them up later.